### PR TITLE
fix(api): Nv 4798 unify default step controls definition using only zod

### PR DESCRIPTION
**Pull request by tatarco**
### What changed? Why was the change needed?

1. Change Control schemas to use Zod as source of truth.
2. Use zod to infer the typescript type within renderers
3. Add digest ZOD & data schema.
4. Add Digest renderer.
5. Add Dealy renderer.
6. Add Delate ZOD
7. Add Delay data schema

### Screenshots

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

File: apps/api/src/app/environments-v1/usecases/output-renderers/digest-output-renderer.usecase.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        },
+      };
+    }
+    if (isDigestTimedControl(parse) && parse.cron) {
+      return {
+        cron: parse.cron,
+        digestKey: parse.digestKey,
+      };
+    }
+    throw new DigestRenderProblem(renderCommand.controlValues);
```
Comment (SokratisVidros) [inline apps/api/src/app/environments-v1/usecases/output-renderers/digest-output-renderer.usecase.ts]:
```suggestion
    throw new ≈(renderCommand.controlValues);
```

Or just reuse one of the existing exceptions. I don't think we need a specific one only for digest.

File: apps/api/src/app/environments-v1/usecases/output-renderers/in-app-output-renderer.usecase.ts [hunk 2]
```diff
@@ -1,87 +1,57 @@
...
-  redirect: z
-    .object({
-      url: z.string().optional(),
-      target: RedirectTargetEnumSchema.optional(), // Optional target
-    })
-    .optional(),
-});
+  private buildRedirect(redirect?: InAppRedirectType) {
+    if (!(redirect && redirect.url)) {
+      return undefined;
```
Comment (SokratisVidros) [inline apps/api/src/app/environments-v1/usecases/output-renderers/in-app-output-renderer.usecase.ts:35]: Is the default undefined or null? I think the FE expected null but I might be wrong.

File: apps/api/src/app/environments-v1/usecases/output-renderers/in-app-output-renderer.usecase.ts [hunk 3]
```diff
@@ -1,87 +1,57 @@
...
+
+    return {
+      url: redirect.url,
+      target: redirect.target as RedirectTargetEnum,
+    };
+  }
+
+  private buildActionIfAllPartsAvailable(primaryAction?: InAppActionType) {
+    if (!(primaryAction && primaryAction.label && primaryAction.redirect && primaryAction.redirect.url)) {
+      return undefined;
```
Comment (SokratisVidros) [inline apps/api/src/app/environments-v1/usecases/output-renderers/in-app-output-renderer.usecase.ts:46]: Ditto about null vs undefined

**Review by SokratisVidros — APPROVED** [review]

File: apps/api/src/app/environments-v1/usecases/output-renderers/in-app-output-renderer.usecase.ts (hunk 2 above)
Comment (tatarco) [reply apps/api/src/app/environments-v1/usecases/output-renderers/in-app-output-renderer.usecase.ts:35]: undefined, it's not a default, it's the outcome of the render.

File: apps/api/src/app/environments-v1/usecases/output-renderers/in-app-output-renderer.usecase.ts (hunk 3 above)
Comment (tatarco) [reply apps/api/src/app/environments-v1/usecases/output-renderers/in-app-output-renderer.usecase.ts:46]: same here, it's not the default, it's the render outcome

**Pull request merged**